sql/src/main/java/org/cubeengine/module/sql/database/impl/SQLDatabase.java

```diff
@@ -162,11 +159,10 @@ else if (dbVersion.isOlderThan(updater.getTableVersion()))
                 {
                     logger.info("table-version is too old! Updating {} from {} to {}", updater.getName(),
                             dbVersion.toString(), version.toString());
-                    try (Connection connection = this.getConnection())
-                    {
-                        updater.update(connection, dbVersion);
-                    }
-                    getDSL().mergeInto(TABLE_VERSION).values(updater.getName(), version.toString());
+                    updater.update(this, dbVersion);
```

Comment:
not sure how well transactional DDL is supported, but that might be interesting to investigate for atomic schema updates.